File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.js
var lengthOfLongestSubstring = function(s) {
   
    let cs = ''
    let m = 0
    
    for(let i = s.length-1; i>= 0; i--){
        const fi = cs.indexOf(s[i])
       if(fi >= 0 ) cs = cs.slice(fi+1)
        cs+=s[i]
       if(cs.length>m) m = cs.length
    }
    
    return m

};

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
    int m = 0;
    int l = 0;
    int r = 0;
    vector<int> c(127);
    while (r < s.length())
    {
        c[s[r]]+=1;
        while (c[s[r]] > 1)
        {
            c[s[l]]--;
            l++;
        }
        m=max(m, r-l+1);
        r++;
    }
    return m;
    }
};